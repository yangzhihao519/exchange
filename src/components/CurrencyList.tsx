import React from 'react'
import Currency from '../models/currency'
import Select from 'react-select'

interface CurrencyListProps {
    allCurrencies: Currency[],
    onSelectCurrency: (selectedCurrencyCode: string, isSellingOut: boolean) => void,
    selectedCurrencyCode: string,
    isSellingOut: boolean
}

interface CurrencyListState {}

class CurrencyList extends React.Component<CurrencyListProps, CurrencyListState>{

    handleOnClick = (e: any) => {
        console.log(e)
        this.props.onSelectCurrency(e.value, this.props.isSellingOut)
    }

    render(){
        let {allCurrencies, selectedCurrencyCode} = this.props
        let allCurrenciesCodes = allCurrencies.map((item) => {return {value: item.code, label: item.code}})
        let selectedCode = {value: selectedCurrencyCode, label: selectedCurrencyCode}

        return <Select
            classNamePrefix='currency-list'
            value={selectedCode}
            onChange={(e:any) => this.handleOnClick(e)}
            options={allCurrenciesCodes}
        />
    }
}

export default CurrencyList