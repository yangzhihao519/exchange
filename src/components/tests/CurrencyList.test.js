import React from 'react';
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import CurrencyList from '../CurrencyList';

const mockFunction = jest.fn();

describe('CurrencyList', () => {
  it('renders without crashing given the required props', () => {
    const props = {
        allCurrencies: [],
        onSelectCurrency: mockFunction,
        selectedCurrencyCode: '',
        isSellingOut: false
    }
    const wrapper = shallow(<CurrencyList {...props} />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})