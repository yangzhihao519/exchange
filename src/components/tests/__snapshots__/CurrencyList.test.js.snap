// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CurrencyList renders without crashing given the required props 1`] = `
<StateManager
  classNamePrefix="currency-list"
  defaultInputValue=""
  defaultMenuIsOpen={false}
  defaultValue={null}
  onChange={[Function]}
  options={Array []}
  value={
    Object {
      "label": "",
      "value": "",
    }
  }
/>
`;
